Packs/IBMSecurityVerify/ModelingRules/IBMSecurityVerify/IBMSecurityVerify.xif

```diff
@@ -0,0 +1,139 @@
+    xdm.event.original_event_type = data -> messageid,
+    xdm.auth.mfa.method = data -> mfamethod,
+    xdm.source.user.identifier = if(data -> performedby IN("System", "shared_apikey"), null, data -> performedby),
+    xdm.source.application.name = coalesce(data -> performedby_clientname, if(data -> performedby IN("System", "shared_apikey"), data -> performedby)),
+    xdm.source.user.username = coalesce(src_user2, data -> username, src_user1, data -> performedby_username), //2 possible username fields, performedby_username and username.
+    xdm.source.user.upn = if(src_user2 != null, data -> username ,src_user1 != null, data -> performedby_username),
+    xdm.target.resource.id = data -> entitlementid,
+    xdm.target.resource.name = data -> entitlementname,
+    xdm.target.resource.sub_type = data -> entitlementsubtype,
+    xdm.target.resource.type = if(data -> entitlementtype != null AND data -> entitlementcategory != null, concat(data -> entitlementtype, " - ", data -> entitlementcategory), coalesce(data -> entitlementtype, data -> entitlementcategory, data -> resource )), // The "type" field represents the role, while the "category" field specifies the role type, e.g., "role - admin."
+    xdm.source.user.groups = to_json_string(data -> entitlements) -> []
+| call generic_fields_mapping;
+
+/*
+Mapping of 'sso' event types.
+*/
+filter event_type = "sso"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.auth.service = data -> subtype,
+    xdm.source.user.identifier = data -> userid,
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.target.application.name = data -> applicationname,
+    xdm.source.host.device_id = data -> deviceid //valid only if its an Android or iPhone device.
+| call generic_fields_mapping;
+
+/*
+Mapping of 'authentication' event types
+*/
+filter event_type = "authentication"
+| alter
+    src_user = arrayindex(regextract(data -> username, "(.+)\@") ,0) // The username can sometimes appear as a username and other times as an email.
+| alter
+    xdm.event.operation_sub_type = coalesce(data -> dict_op, data -> action),
+    xdm.event.description = data -> cause,
+    xdm.auth.service = data -> sourcetype,
+    xdm.source.user.identifier = if(data -> subject = "UNKNOWN", null, coalesce(data -> subject, data -> performedby)),
+    xdm.target.url = data -> ["target"],
+    xdm.source.user.username = coalesce(src_user, data -> username),
+    xdm.source.user.upn = if(src_user != null, data -> username),
+    xdm.source.host.device_id = data -> deviceid,
+    xdm.auth.mfa.method = data -> mfamethod,
+    xdm.auth.mfa.client_details = data -> mfadevice,
+    xdm.event.original_event_type = coalesce(data -> messageid, data -> messageId),
+    xdm.session_context_id = data -> usersessionid
```

Comment:
I think that correlationid is better mapping for that - we should understand what is the best field that describes a "session" in order to correlate several events into a session

Reply:
@yassorpanw 
I think that we should use usersessionid field, because comparing to correlationid, there are events with the same usersessionid while correlationid is unique for every event (in sso and authentication events). 

The only downside is that usersessionid has null values for some events, but I think that this is the that should be use to correlate events under the same authentication flow.

If you agree, we can resolve this conversation.